gio/src/subclass/list_model.rs

```diff
@@ -0,0 +1,92 @@
+use crate::ListModel;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, IsA, ObjectExt};
+use once_cell::sync::Lazy;
+
+pub trait ListModelImpl: ObjectImpl {
+    fn get_item_type(&self, list_model: &Self::Type) -> glib::Type;
+    fn get_n_items(&self, list_model: &Self::Type) -> u32;
+    fn get_item(&self, list_model: &Self::Type, position: u32) -> Option<glib::Object>;
+}
+
+unsafe impl<T: ListModelImpl> IsImplementable<T> for ListModel
+where
+    <T as ObjectSubclass>::Type: IsA<glib::Object>,
```

Comment:
As `IOStream` works without those `where` clauses, they're probably not actually needed here? I mean, the wrapper we get down there is actually a `gio::ListModel`.

Reply:
The difference is `IOStream` is not an interface, so it doesn't needs the extra conditions as imposed by the `where` clauses

Reply:
Ah of course. Makes sense :)